### DISCUSSION TITLE
WebAssembly Azure AppService puzzle

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#1980] [Q&A] [answered]
I have an Oqtane app running in an Azure Appservice.  I'd like to change it to Runtime="WebAssembly" so I've edited the appsettings.json in wwwroot using the Kudu tools.  Then I restart the appservice expecting the app to change.  When I hard-refresh after the appstart, I'm still getting browse conole messages about sockets, and even socket errors. 

Are there additional settings or deployment steps needed to switch to Runtime="WebAssembly"?

**Comment by leigh-pointer** [comment, marked as answer]
I think you need to change it in Site Settings in the Hosting Model section

![image](https://user-images.githubusercontent.com/9447612/152619905-a79ac97b-4600-4d84-b364-0ce4689b81a5.png)

**Reply by markdav-is** [reply]
thanks for the tip!  I was going for the appsettings.json.  I think we'll need to pump up the module configs before we flip the switch to grab all the dependencies.

**Comment by sbwalker** [comment]
@markdav-is I am hoping @leigh-pointer  was able to point you in the right direction. The framework was enhanced so that you can specify your preferred Blazor hosting model for each Site in your installation. The appsetting is only used as a default during the initial installation and as a default when creating new Sites.
